src/gui/editorwidgets/qgsmultiedittoolbutton.cpp

```diff
@@ -0,0 +1,126 @@
+/***************************************************************************
+     qgsmultiedittoolbutton.cpp
+     --------------------------
+    Date                 : March 2016
+    Copyright            : (C) 2016 Nyall Dawson
+    Email                : nyall dot dawson at gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmultiedittoolbutton.h"
+#include "qgsapplication.h"
+#include <QMenu>
+
+QIcon QgsMultiEditToolButton::sIconInitialState;
```

Comment:
static QIcon's are incompatible with Qt5. They will be cleaned after QApplication has been deleted and crash on exit.

See also:

https://github.com/qgis/QGIS/pull/2996/commits/286e0de3987efad282080f905b236e72d210c262